### PR TITLE
Open source deeplab changes. Check change log in README.md for detailed info.

### PR DESCRIPTION
**Pull request by huihui-personal**

### CODE REVIEW

**Comment by huihui-personal** [comment]
@aquariusjay @YknZhu Pls help resolve the conflict, it should be raw_prediction.

**Comment by aquariusjay** [comment]
Thanks, Hui Hui!
Could you please help run a few tests to make sure the code is correct?
In particular, common_test.py, model_test.py, local_test.sh, and local_test_mobilenetv2.sh?

**Comment by huihui-personal** [comment]
There are some bugs when running the scripts. I have fixed all of them and all the scripts are passing now.

File: research/deeplab/deeplab_demo.ipynb [hunk 1]
```diff
@@ -294,13 +294,13 @@
         "  try:\n",
         "    f = urllib.request.urlopen(url)\n",
         "    jpeg_str = f.read()\n",
-        "    original_im = Image.open(BytesIO(jpeg_str))\n",
+        "    orignal_im = Image.open(BytesIO(jpeg_str))\n",
```
Comment (aquariusjay) [inline research/deeplab/deeplab_demo.ipynb]: original_im
I think this is a typo. :)

File: research/deeplab/deeplab_demo.ipynb [hunk 2]
```diff
@@ -294,13 +294,13 @@
...
         "  except IOError:\n",
         "    print('Cannot retrieve image. Please check url: ' + url)\n",
         "    return\n",
         "\n",
         "  print('running deeplab on image %s...' % url)\n",
-        "  resized_im, seg_map = MODEL.run(original_im)\n",
+        "  resized_im, seg_map = MODEL.run(orignal_im)\n",
         "\n",
```
Comment (aquariusjay) [inline research/deeplab/deeplab_demo.ipynb]: original_im

File: research/deeplab/core/nas_cell.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    """Concatenates the unused hidden states of the cell."""
+    used_hiddenstates = self._used_hiddenstates
+    states_to_combine = ([
+        h for h, is_used in zip(net, used_hiddenstates) if not is_used])
+    net = tf.concat(values=states_to_combine, axis=3)
+    return net
+
+  @tf.contrib.framework.add_arg_scope  # No public API. For internal use only.
+  def _apply_drop_path(self, net):
+    """Apply drop_path regularization."""
```
Comment (YknZhu) [inline research/deeplab/core/nas_cell.py]: nit: Applies

File: research/deeplab/core/nas_cell.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+  def _combine_unused_states(self, net):
+    """Concatenates the unused hidden states of the cell."""
+    used_hiddenstates = self._used_hiddenstates
+    states_to_combine = ([
+        h for h, is_used in zip(net, used_hiddenstates) if not is_used])
+    net = tf.concat(values=states_to_combine, axis=3)
+    return net
+
+  @tf.contrib.framework.add_arg_scope  # No public API. For internal use only.
```
Comment (YknZhu) [inline research/deeplab/core/nas_cell.py]: remove

Comment (YknZhu) [reply research/deeplab/core/nas_cell.py]: ahha sorry I meant remove the comment... add_arg_scope is a valid public API

**Pull request merged**